src/utilities/assertValidName.js

```diff
@@ -8,7 +8,7 @@
  *  of patent rights can be found in the PATENTS file in the same directory.
  */
 
-const NAME_RX = /^[_a-zA-Z][_a-zA-Z0-9]*$/;
+const NAME_RX = /^[_a-zA-Z][_a-zA-Z-0-9]*$/;
```

Comment:
This is a pretty treacherous way to allow a hyphen, buried in the middle like that. Would be better to put the hyphen at the very beginning or end of the range; eg:

```
const NAME_RX = /^[_a-zA-Z][_a-zA-Z0-9-]*$/;
```